dpcpp/CMakeLists.txt

```diff
@@ -51,11 +51,16 @@ target_sources(ginkgo_dpcpp
 
 ginkgo_compile_features(ginkgo_dpcpp)
 
-set(GINKGO_DPCPP_FLAGS ${GINKGO_COMPILER_FLAGS} -fsycl -fsycl-device-lib=libc)
+set(GINKGO_DPCPP_FLAGS ${GINKGO_COMPILER_FLAGS} -fsycl)
 set(GINKGO_DPCPP_FLAGS ${GINKGO_DPCPP_FLAGS} PARENT_SCOPE)
 target_compile_options(ginkgo_dpcpp PRIVATE "${GINKGO_DPCPP_FLAGS}")
 target_compile_features(ginkgo_dpcpp PRIVATE cxx_std_17)
+target_link_options(ginkgo_dpcpp PRIVATE -fsycl-device-lib=all)
+target_link_options(ginkgo_dpcpp PRIVATE -fsycl-device-code-split=per_kernel)
```

Comment:
dpcpp also provides ahead-of-time (AOT) to select the device and compile the assembly to eliminate the JIT time.
but I need to check whether AOT allowed some unused illegal subgroup function existing.

Reply:
I think that mode will be quite complicated to setup, particularly since there is nothing to help set it up in CMake, thus it would require architecture detection like the CAS and possibly other additions as well?

Reply:
Yes, It should need more changes. the current version is enough for our usage. 
AOT may be helpful to detect error for those devices we do not have or eliminate the JIT time in the future